src/syscheckd/src/recovery/recovery.cpp

```diff
@@ -0,0 +1,154 @@
+#include <cstdint>
+#include <vector>
+#include <string>
+#include <hashHelper.h>
+
+extern "C" {
+    #include "debug_op.h" // Use the same C-style logging as the rest of FIM's code.
+}
+
+#include "recovery.h"
+#include "agent_sync_protocol_c_interface.h"
+#include "agent_sync_protocol_c_wrapper.hpp"
+#include "db.hpp"
+#include "fimCommonDefs.h"
+#include "stringHelper.h"
+#include "ipersistent_queue.hpp"
+#include "syscheck.h"
+#include <chrono>
+
+/**
+ * @brief Get the current timestamp in UNIX format
+ */
+int64_t getUnixTimeSeconds() {
+    return std::chrono::duration_cast<std::chrono::seconds>(
+        std::chrono::system_clock::now().time_since_epoch()
+    ).count();
+}
+
+/**
+ * @brief Calculate the checksum-of-checksums for a table
+ * @param table_name The table to calculate checksum for
+ * @return The SHA1 checksum-of-checksums as a hex string
+ */
+std::string calculateTableChecksum(const char* table_name) {
```

Comment:
Could this method be part of the DB class and placed alongside DB::getConcatenatedChecksums?